rust/src/shards/gui/widgets/hex_viewer.rs

```diff
@@ -0,0 +1,192 @@
+    Ok(data.inputType)
+  }
+
+  fn requiredVariables(&mut self) -> Option<&ExposedTypes> {
+    self.requiring.clear();
+
+    // Add UI.Parents to the list of required variables
+    util::require_parents(&mut self.requiring, &self.parents);
+
+    Some(&self.requiring)
+  }
+
+  fn warmup(&mut self, context: &Context) -> Result<(), &str> {
+    self.parents.warmup(context);
+
+    Ok(())
+  }
+
+  fn cleanup(&mut self) -> Result<(), &str> {
+    self.parents.cleanup();
+
+    Ok(())
+  }
+
+  fn activate(&mut self, _context: &Context, input: &Var) -> Result<Var, &str> {
+    use egui_memory_editor::MemoryEditor;
+
+    if let Some(ui) = util::get_current_parent(*self.parents.get())? {
+      let mem = unsafe {
+        let (data, len) = match input.valueType {
+          SHType_Bytes => (
+            input.payload.__bindgen_anon_1.__bindgen_anon_4.bytesValue,
+            input.payload.__bindgen_anon_1.__bindgen_anon_4.bytesSize as usize,
+          ),
+          SHType_Enum => (
+            &input.payload.__bindgen_anon_1.__bindgen_anon_3.enumValue as *const i32 as *mut u8,
+            4,
+          ),
+          SHType_Float => (
+            &input.payload.__bindgen_anon_1.floatValue as *const f64 as *mut u8,
+            8,
+          ),
+          SHType_Float2 => (
+            &input.payload.__bindgen_anon_1.float2Value as *const f64 as *mut u8,
+            16,
+          ),
+          SHType_Float3 => (
+            &input.payload.__bindgen_anon_1.float3Value as *const f32 as *mut u8,
+            12,
+          ),
+          SHType_Float4 => (
+            &input.payload.__bindgen_anon_1.float4Value as *const f32 as *mut u8,
+            16,
+          ),
+          SHType_Int => (
+            &input.payload.__bindgen_anon_1.intValue as *const i64 as *mut u8,
+            8,
+          ),
+          SHType_Int2 => (
+            &input.payload.__bindgen_anon_1.int2Value as *const i64 as *mut u8,
+            16,
+          ),
+          SHType_Int3 => (
+            &input.payload.__bindgen_anon_1.int3Value as *const i32 as *mut u8,
+            12,
+          ),
+          SHType_Int4 => (
+            &input.payload.__bindgen_anon_1.int4Value as *const i32 as *mut u8,
+            16,
+          ),
+          SHType_Int8 => (
+            &input.payload.__bindgen_anon_1.int8Value as *const i16 as *mut u8,
+            16,
+          ),
+          SHType_Int16 => (
+            &input.payload.__bindgen_anon_1.int16Value as *const i8 as *mut u8,
+            16,
+          ),
+          SHType_String => (
+            input.payload.__bindgen_anon_1.__bindgen_anon_2.stringValue as *mut u8,
+            input.payload.__bindgen_anon_1.__bindgen_anon_2.stringLen as usize,
+          ),
+          _ => unreachable!(),
+        };
+        std::slice::from_raw_parts_mut(data, len)
+      };
+
+      let mut mem_editor = MemoryEditor::new().with_address_range("All", 0..mem.len());
```

Comment:
anything new is hyper misleading, this is just the way UI api works (badly) right? doesn't mean heavy allocations...

Reply:
`with_address_range` only available during construction: it moves `Self` (doesn't return a reference), so if we keep a reference to the `MemoryEditor`, we can't modify the range afterwards. It's a limitation of this library currently.

Reply:
Issue opened: https://github.com/Hirtol/egui_memory_editor/issues/5

Reply:
PR created: Hirtol/egui_memory_editor#6